solr/core/src/java/org/apache/solr/schema/ScalarQuantizedDenseVectorField.java

```
@@ -0,0 +1,154 @@
+    this.bits = ofNullable(args.remove(BITS_PARAM)).map(Integer::parseInt).orElse(DEFAULT_BITS);
+
+    this.compress =
+        ofNullable(args.remove(COMPRESS_PARAM)).map(Boolean::parseBoolean).orElse(false);
+
+    this.confidenceInterval =
+        ofNullable(args.remove(CONFIDENCE_INTERVAL_PARAM))
+            .map(Float::parseFloat)
+            .orElse(DEFAULT_CONFIDENCE_INTERVAL);
+
+    if (ofNullable(args.remove(DYNAMIC_CONFIDENCE_INTERVAL_PARAM))
+        .map(Boolean::parseBoolean)
+        .orElse(false)) {
+      this.confidenceInterval = Lucene99ScalarQuantizedVectorsFormat.DYNAMIC_CONFIDENCE_INTERVAL;
+    }
+
+    super.init(schema, args);
+  }
+
+  @Override
+  public KnnVectorsFormat buildKnnVectorsFormat() {
+    return new Lucene99HnswScalarQuantizedVectorsFormat(
+        getHnswMaxConn(),
+        getHnswBeamWidth(),
+        DEFAULT_NUM_MERGE_WORKER,
+        getBits(),
+        useCompression(),
+        getConfidenceInterval(),
+        null);
+  }
+
+  @Override
+  public void checkSchemaField(final SchemaField field) throws SolrException {
+    super.checkSchemaField(field);
+
+    if (confidenceInterval != null
+        && confidenceInterval != Lucene99ScalarQuantizedVectorsFormat.DYNAMIC_CONFIDENCE_INTERVAL
+        && (confidenceInterval < MINIMUM_CONFIDENCE_INTERVAL
+            || confidenceInterval > MAXIMUM_CONFIDENCE_INTERVAL)) {
+      throw new SolrException(
+          SolrException.ErrorCode.SERVER_ERROR,
+          getClass().getSimpleName()
+              + " fields must have non-dynamic confidence interval between 0.9 and 1.0 "
+              + field.getName());
+    }
+    if (getBits() != 4 && getBits() != 7) {
+      throw new SolrException(
+          SolrException.ErrorCode.SERVER_ERROR,
+          getClass().getSimpleName()
+              + " fields must have bit size of 4 (half-byte) or 7 (signed-byte) "
+              + field.getName());
+    }
+
+    if (useCompression() && getBits() != 4) {
```

Can't we rely on "if (bits > 4 && compress) {
      // compress=true otherwise silently does nothing when bits=7?
      throw new IllegalArgumentException("compress=true only applies when bits=4");
    }" Lucene side?